whylog/config/parsers.py

```diff
@@ -57,3 +67,136 @@ def create_from_intent(cls, parser_intent):
     @classmethod
     def from_dao(cls, serialized_parser):
         return RegexParser(**serialized_parser)
+
+
+class ConcatedRegexParser(object):
+    NO_MATCH = frozendict()
+
+    def __init__(self, parser_list):
+        self._parsers = parser_list
+        if IMPORTED_RE:
+            return
+        forward, backward = self._create_concated_regexes()
+        self._forward_regex = regex.compile(forward)
+        self._backward_regex = regex.compile(backward)
+        self._forward_parsers_indexes = self._get_indexes_of_groups_for_parsers(self._parsers)
+        self._backward_parsers_indexes = self._get_indexes_of_groups_for_parsers(
+            reversed(
+                self._parsers
+            )
+        )
+        self._numbers_in_list = self._number_in_list()
+        self._forward_group_index_to_regex = self._create_group_index_to_regex_name(
+            self._forward_parsers_indexes
+        )
+        self._backward_group_index_to_regex = self._create_group_index_to_regex_name(
+            self._backward_parsers_indexes
+        )
+
+    def _create_concated_regexes(self):
+        forward_regex = "|".join("(" + parser.regex_str + ")" for parser in self._parsers)
+        backward_regex = "|".join(
+            "(" + parser.regex_str + ")" for parser in reversed(self._parsers)
+        )
+        return forward_regex, backward_regex
+
+    def _get_indexes_of_groups_for_parsers(self, parser_list):
+        indexes_dict = {}
+        free_index = 0
+        for parser in parser_list:
+            amount_of_group = parser.regex_str.count('(') - parser.regex_str.count('\(')
+            indexes_dict[parser.name] = (free_index, amount_of_group)
+            free_index += amount_of_group + 1
+        return indexes_dict
+
+    def _number_in_list(self):
+        regex_numbers = {}
+        number = 0
+        for parser in self._parsers:
+            regex_numbers[parser.name] = number
+            number += 1
+        return regex_numbers
+
+    def _create_group_index_to_regex_name(self, parsers_indexes):
+        index_to_regex = {}
+        for name, indexes in parsers_indexes.items():
+            index_to_regex[indexes[0]] = name
+        return index_to_regex
+
+    def get_extracted_regex_params(self, line):
```

Comment:
it returns clues, not regex params, so the method should be called appropriately